R/nm-file.R

```diff
@@ -115,16 +122,33 @@ nm_data <- function(.mod, .sep = ",") {
 # PRIVATE HELPERS AND IMPLEMENTATION
 
 #' Implementation function for reading NONMEM files
-#' @importFrom readr read_table read_table2 cols read_lines
+#' @importFrom tibble as_tibble
+#' @importFrom data.table fread
 #' @importFrom stringr str_detect
 #' @keywords internal
 nm_file_impl <- function(.path, .est_method) {
   # read file and find top of table
   verbose_msg(glue("Reading {basename(.path)}"))
-  .txt <- read_lines(.path)
-  .est <- which(str_detect(.txt, "^ *TABLE NO"))
+
+  # get line numbers of TABLE lines
+  checkmate::assert_file_exists(.path)
+  .est <- processx::run(
```

Comment:
I don't think we need to do this every time. Let's talk about my previous comments and how to determine this.

Reply:
I timed it on some pretty big files (100k lines, I think) and it was super quick. I hear your point, but I think any other solution is going to end up being more complicated than it's worth, especially given that this doesn't really cause problems.

Reply:
What is the use case here? Why are you looking?

Reply:
although, @dpastoor I meant to have a conversation about how we should do this on Windows (or any system without `grep` I guess). Any thoughts? Our "support" for windows is a bit thin, but at the very least we should give a reasonable warning or something. Thoughts?

Reply:
@seth127 answer this question first: why are you looking through the file in the first place?

Reply:
to see if there's more than one table in it

Reply:
Ok, good.  Why might there be more than one table there? .... in this use case?

Reply:
The main case where you might see multiple tables would be a simulation run; we should refuse to process this and can be easily checked from the names in the control stream. As you note, someone could (possibly) could write the same table file by mistake (we can say "mistake" b/c I don't think anyone is really doing this intentionally). This is an easy check of duplicate table file names. Once you got that far, I think you're in the clear and you don't need to be grepping the file every time. I think grepping the file is great when you have a large simulation run to process but we already took that option off the table, right?  

I'll put it another way: this function is to make more convenient something that we are already doing: reading in the data and a straightforward read in of the table file (with fread or readr). We are already doing it in this straightforward, read-the-table way and I wanted to capture that workflow.  

I can't guarantee that no one will ever come across a problematic situation where reading the table gets mangled. If you wanted to be super careful to exit in a nice way no matter what, then wrap in `try()`, let the user know something failed (unlikely IMO if we rule out $SIM and duplicate table files), and ask them to go back and ready the intended use of the function.